src/components/message/message-card-list/MessagesVirtualizedList.tsx

```diff
@@ -95,26 +95,7 @@ const MessagesVirtualizedList = (props: Props) => {
 	]);
 
 	React.useEffect(() => {
-		toJS(messages).forEach(message =>
-			message.parsedMessages
-				? message.parsedMessages.forEach((parsedMessage, index) => {
-						const tempMessage = message;
-						const { parsedMessages, ...rest } = tempMessage;
-						const tempMessageItem: EventMessageItem = {
-							...rest,
-							parsedMessage: null,
-							parsedMessages: [],
-						};
-
-						tempMessageItem.parsedMessage = tempMessage.parsedMessages
-							? tempMessage.parsedMessages[index]
-							: null;
-						if (tempMessageItem.parsedMessages && tempMessageItem.parsedMessage)
-							tempMessageItem.parsedMessages[0] = tempMessageItem.parsedMessage;
-						setMessageList(messageListCopy => [...messageListCopy, tempMessageItem]);
-				  })
-				: setMessageList(messageListCopy => [...messageListCopy, message as EventMessageItem]),
-		);
+		setMessageList(messageListCopy => [...messageListCopy, ...getMessageList(messages)]);
```

Comment:
1. this code will create lots of duplicate messages.
2. this useEffect will trigger each time we add a message to message list. e.g. we had 1000 messages, we added 1. and still we will have to loop over whole array. can we do better?